pynitrokey/cli/nk3/__init__.py

```diff
@@ -789,3 +794,102 @@ def clear_password(ctx: Context, experimental: bool) -> None:
             local_print(
                 f"Device returns error: {e}. This passphrase might be invalid or is set already."
             )
+
+
+@nk3.group(hidden=True)
+def provision() -> None:
+    """
+    Provision the device.  This command is only used during the production
+    process and not available for regular devices.
+    """
+    pass
+
+
+@provision.command("fido2")
+@click.pass_obj
+@click.option(
+    "--key",
+    "key_file",
+    required=True,
+    type=click.File("rb"),
+    help="The path of the FIDO2 attestation key",
+)
+@click.option(
+    "--cert",
+    "cert_file",
+    required=True,
+    type=click.File("rb"),
+    help="The path of the FIDO2 attestation certificate",
+)
+def provision_fido2(ctx: Context, key_file: BinaryIO, cert_file: BinaryIO) -> None:
+    """Provision the FIDO2 attestation key and certificate."""
+    key = key_file.read()
+    cert = cert_file.read()
+
+    if len(key) != 36:
+        raise CliException(f"Invalid key length {len(key)} (expected 36)")
+    ecdsa_key = SigningKey.from_string(key[4:], curve=NIST256p)
+    pem_pubkey = serialization.load_pem_public_key(
+        ecdsa_key.get_verifying_key().to_pem()
+    )
+
+    x509_cert = x509.load_der_x509_certificate(cert)
+    cert_pubkey = x509_cert.public_key()
+
+    if not isinstance(pem_pubkey, EllipticCurvePublicKey):
+        raise CliException("The FIDO2 attestation key is not an EC key")
+    if not isinstance(cert_pubkey, EllipticCurvePublicKey):
+        raise CliException(
+            "The FIDO2 attestation certificate does not contain an EC key"
+        )
+    if pem_pubkey.public_numbers() != cert_pubkey.public_numbers():
+        raise CliException(
+            "The FIDO2 attestation certificate does not match the public key"
+        )
+
+    # See https://www.w3.org/TR/webauthn-2/#sctn-packed-attestation-cert-requirements
+    if x509_cert.version != x509.Version.v3:
+        raise CliException(
+            f"Unexpected certificate version {x509_cert.version} (expected v3)"
+        )
+
+    found_c = False
+    found_o = False
+    found_cn = False
+    for attr in x509_cert.subject:
+        if attr.rfc4514_attribute_name == "C":
+            found_c = True
+        elif attr.rfc4514_attribute_name == "O":
+            found_o = True
+        elif attr.rfc4514_attribute_name == "CN":
+            found_cn = True
+        elif attr.rfc4514_attribute_name == "OU":
+            if attr.value != "Authenticator Attestation":
+                raise CliException(
+                    f"Unexpected certificate subject OU {attr.value} (expected "
+                    "Authenticator Attestation)"
+                )
+    if not found_c:
+        raise CliException("Missing subject C in certificate")
+    if not found_o:
+        raise CliException("Missing subject O in certificate")
+    if not found_cn:
+        raise CliException("Missing subject CN in certificate")
```

Comment:
nit: style - I think this could be better realized by a dynamic dict, and testing for the presence of the values of the keys directly:
```python
d = defaultdict(lambda: None)
for attr in x509_cert.subject:
    d[attr.rfc4514_attribute_name] = attr
if not (d["C"] and d["O"] and ...):
    raise

# or even

req_ids = ["O", "CN", ...]
if not all( [ d[x] for x in req_ids  ] ):
    raise

```
Obviously this is a quick implementation (and I had taken more time to review code probably), so no need to correct that, but maybe worth considering next time.

Reply:
Good point.  I was so fed up with `cryptography` not providing a nice API to query the attribute by name that I just went into brute-force mode. ;)

Reply:
Fixed.  I’ve also added the cert hash to the output for easier interpretation of provisioning logs.